i18n4k-generator/src/main/kotlin/de/comahe/i18n4k/generator/I18n4kGenerator.kt

```diff
@@ -179,44 +153,132 @@ class I18n4kGenerator(
             }
         }
 
+        fun addFactoryClass(className: ClassName, key: String, index: Int, params: Map<String, String?>) {
+            val messageBundle = "messageBundle"
+            val pkg = "de.comahe.i18n4k"
+            val messageBundleType = ClassName("$pkg.messages", messageBundle.capitalize())
+            val reserved = listOf("val", "var", "class", "package", "this", "when", "in", "is")
+            val paramsKeyString = params.keys.joinToString { if (it in reserved) "`$it`" else it }
+            fun FunSpec.Builder.addParams(): FunSpec.Builder = apply {
+                for ((name, type) in params) {
+                    val t = when(type?.lowercase()) {
+                        "string" -> String::class
+                        "int" -> Int::class
+                        else -> Any::class
+                    }.asClassName()
+                    addParameter(name, t)
+                }
+            }
+            fun FunSpec.Builder.addLocale(): FunSpec.Builder = apply {
+                val param = ParameterSpec
+                    .builder("locale", Locale::class.asClassName().copy(nullable = settings.globalLocaleAsDefault))
```

Comment:
`Locale::class.asClassName()` this seems to lead to such strange import declaration with is not multi platform compatible.

```kt
import de.comahe.i18n4k.Locale as I18n4kLocale
import java.util.Locale as UtilLocale
```

Please use  `val localeClassName = ClassName("de.comahe.i18n4k", "Locale")`. The instance in `generateLocalisationObject` may become a class member.